plugins/httpapi/aci.py

```diff
@@ -0,0 +1,218 @@
+# Copyright (c) 2022 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+author:
+- Shreyas Srish (shrsr)
+httpapi: aci
+short_description: Ansible ACI HTTPAPI Plugin.
+description:
+  - This ACI plugin provides the HTTPAPI transport methods needed to initiate
+    a connection to the APIC, send API requests and process the
+    response from the controller.
+"""
+
+import json
+import re
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.connection import ConnectionError
+from ansible.plugins.httpapi import HttpApiBase
+
+
+class HttpApi(HttpApiBase):
+
+    def __init__(self, *args, **kwargs):
+        super(HttpApi, self).__init__(*args, **kwargs)
+        self.auth = None
+        self.check_auth_from_private_key = None
+        self.backup_hosts = None
+        self.list_of_hosts = []
+        self.host_counter = 0
+        self.entered_exception = False
+        
+    def set_params(self, auth, params):
+        self.params = params
+        self.auth = auth
+
+    def get_backup_hosts(self):
+        try:
+            # append is used here to store the first value of the variable self.connection.get_option("host") in the 0th position
+            # this is done because value of 'host' keeps changing constantly when list of hosts is provided. We always
+            # want access to the original set of hosts
+            self.list_of_hosts.append(re.sub(r'[[\]]', '', self.connection.get_option("host")).split(","))
```

Comment:
should we enforce set instead of list? since they should be unique values or do we allow multiple same hosts?

Reply:
ok makes sense, do we need to filter out duplicate values then from the list?